samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al

```diff
@@ -416,13 +442,16 @@ codeunit 50171 AzureBlobStorage
         end;
 
         // Add Authorization header
+        resourcePath := TypeHelper.UriEscapeDataString(resourcePath).Replace('%2F', '/');
+
         token := GetSasToken(methodVerb, ResourcePath, UtcDateTime, ContentLength, ContentType, CreateCanonicalizedParameters(parameters), xHeaderList);
         headers.Add('Authorization', token);
 
         request.Method := methodVerb;
         if parameters <> '' then
-            resourcePath := resourcePath + '?' + parameters;
-        request.SetRequestUri(ResourceUri + ResourcePath);
+            resourcePath := resourcePath + '?' + parameters.Replace('/', '%2F');
```

Comment:
It is probably just the value part of the parameters that should be encoded using UriEscapeDataString(). Do you have an example where this is needed?

Reply:
This is needed to support a prefix with folder and subfolder support: ListBlobs -Prefix container/MainFolder/SubFolder/Document-